tests/unit/pubsub_v1/subscriber/test_streaming_pull_manager.py

```diff
@@ -144,6 +144,15 @@ def test_ack_deadline():
     assert manager.ack_deadline == 20
 
 
+def test_ack_deadline_w_max_extension_period():
+    manager = make_manager()
+    manager._flow_control = types.FlowControl(max_extension_period=5)
+    for _ in range(5):
+        manager.ack_histogram.add(20)
+
+    assert manager.ack_deadline == 5
```

Comment:
Setting ack time to `20` five times in a row. With this 99% percentile will be near `20`, so `max_extension_period` must be chosen as ack deadline